contracts/AaveAdapter.sol

```diff
@@ -0,0 +1,182 @@
+            amountsRedeemed[0] = ILendingPool(to).withdraw(
+                tokens[0],
+                amounts[0],
+                onBehalfOf
+            );
+        }
+        return amountsRedeemed;
+    }
+
+    // ethIndex is index of ETH amount in amounts array - 255 means no eth
+    function borrow(
+        address onBehalfOf,
+        address[] calldata tokens,
+        uint256[] calldata amounts,
+        address to,
+        uint8 ethIndex
+    ) external override returns (uint256[] memory amountsTransferred) {
+        require(
+            tokens.length == 1 && amounts.length == 1,
+            "Aave supports only single token action at a time"
+        );
+
+        if (ethIndex < 255) {
+            ILendingPool(to).borrow(
+                address(wMATIC),
+                amounts[0],
+                2,
+                0,
+                onBehalfOf
+            );
+            IWETH(wMATIC).withdraw(amounts[0]);
+            _safeTransferETH(onBehalfOf, amounts[0]);
+        } else {
+            //TODO make it flexible to choose interest rate mode
+            ILendingPool(to).borrow(tokens[0], amounts[0], 2, 0, onBehalfOf);
+            IERC20(tokens[0]).transfer(onBehalfOf, amounts[0]);
+        }
+        return amounts;
+    }
+
+    // ethIndex is index of ETH amount in amounts array - 255 means no eth
+    function repay(
+        address onBehalfOf,
+        address[] calldata tokens,
+        uint256[] calldata amounts,
+        address to,
+        uint8 ethIndex
+    ) external payable override returns (uint256[] memory amountsTransferred) {
+        require(
+            tokens.length == 1 && amounts.length == 1,
+            "Aave supports only single token action at a time"
+        );
+
+        uint256[] memory amountsRepayed = new uint256[](1);
+        if (ethIndex < 255) {
+            IWETH(wMATIC).approve(to, amounts[0]);
+            IWETH(wMATIC).deposit{value: amounts[0]}();
+            ILendingPool(to).repay(address(wMATIC), amounts[0], 2, onBehalfOf);
+
+            // refund remaining dust eth/matic
+            if (msg.value > amounts[0]) {
+                _safeTransferETH(onBehalfOf, msg.value - amounts[0]);
+            }
+        } else {
+            IERC20(tokens[0]).transferFrom(
+                onBehalfOf,
+                address(this),
+                amounts[0]
+            );
+            IERC20(tokens[0]).approve(to, amounts[0]);
+            amountsRepayed[0] = ILendingPool(to).repay(
+                tokens[0],
+                amounts[0],
+                2,
+                onBehalfOf
+            );
+        }
+        return amountsTransferred;
+    }
+
+    /**
+     * @dev transfer ETH/Matic to an address, revert if it fails.
+     * @param to recipient of the transfer
+     * @param value the amount to send
+     */
+    function _safeTransferETH(address to, uint256 value) internal {
+        (bool success, ) = to.call{value: value}(new bytes(0));
```

Comment:
may be better to use `address payable x = address(to); x.send(value);`. What do you think?